### ISSUE TITLE
Resource estimate executive summary lacks BRAM numbers

### ISSUE THREAD

**Issue by anshumanmohan** [calyxir/calyx#2288]
_[Synthesis 101](https://github.com/cucapra/notes/blob/main/synthesis.md)_ is a short guide about generating syntheses from .futil files. In refreshing it, I noticed that the executive summary, generated using the command:
```
fud e --to resource-estimate <filename>.futil
```
makes no mention of BRAMs. If you generate the actual report using:
```
fud e --to synth-files -o report <filename>.futil
```
the report has, deep in its guts, the appropriate numbers. 

If you digest the `report` directory into an executive summary using:
```
fud e --from synth-files --to resource-estimate report
```
the executive summary again lacks the BRAM numbers.

So: it's all in there, and it's not a `fud` chaining issue. Some script is just failing to grab and display the BRAM numbers.

**Comment by anshumanmohan** [comment]
cc @polybeandip in case you have anything to add!

**Comment by anshumanmohan** [comment]
Heh I have tried to give this issue an appropriate label, but nothing seems to fit. Sorry Rachit, I know those are important

**Comment by polybeandip** [comment]
Not much to add from me!

I guess in case anyone wants to tackle this issue, I think function to be tweaked is place_and_route_extract` in [`extract.py`](fud/fud/stages/vivado/extract.py). Not fully sure though

**Comment by rachitnigam** [comment]
Yup, you need to find the right table using the parser’s API and then extract the information!

**Comment by polybeandip** [comment]
Just to clarify, is this number from `report/out/FutilBuild.runs/impl_1/main_utilization_placed.rpt` what we want?
![bram](https://github.com/user-attachments/assets/cbd26149-044f-4fa5-8f2b-74e842063664)

**Comment by rachitnigam** [comment]
Yup, looks like the right place to me. It is surprising and perhaps a little concerning that no BRAMs are used. Maybe the internal memories used in FIFOs are not getting mapped correctly?

OTOH, John Demme did mention that LUTRAMs often get used to map FIFOs...
